core/src/main/scala/kafka/server/LogOffsetMetadata.scala

```diff
@@ -39,8 +39,8 @@ object LogOffsetMetadata {
  *  3. the physical position on the located segment
  */
 case class LogOffsetMetadata(messageOffset: Long,
-                             segmentBaseOffset: Long = Log.UnknownOffset,
-                             relativePositionInSegment: Int = LogOffsetMetadata.UnknownFilePosition) {
+                             val segmentBaseOffset: Long = Log.UnknownOffset,
```

Comment:
nit: I think `val` is redundant for case classes.

Reply:
Yes. Jose mentioned this too. Removed.